README.md
  
> 
>    Don't Forget To Update Covid19-Cli Frequently, Thank You For Using

>    Use `pip install -U covid19-cli` or `pip install covid19-cli==x.y.z` x.y.z is the latest version
>

## Covid19-Cli

<P>
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A CLI application for getting covid-19 coronavirus :space_invader:	 status of your country, state or district at lightning :zap: speed right on your terminal :computer: built using https://api.covid19india.org.
</p>

<p>

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Made with :heavy_heart_exclamation: using Python3.

[![PyPI version](https://badge.fury.io/py/covid19-cli.svg)](https://badge.fury.io/py/covid19-cli) ![PyPI - Downloads](https://img.shields.io/pypi/dm/covid19-cli)

</p>

![image](https://github.com/asprazz/covid19-cli/blob/master/screenshots/screenshot4.png)


### Dependencies
- `Covid19` runs on Python3.x
- [`pip3`](https://pip.pypa.io/en/stable/installing/)
- [`requests`](https://requests.readthedocs.io/en/master/user/install/)
- [`argparse`](https://pypi.org/project/argparse/)
- [`prettytable`](https://ptable.readthedocs.io/en/latest/installation.html)
and our beloved
- [`colorama`](https://pypi.org/project/colorama/)
- Thanks to all :pray:


### Installation (Not for development)
- <strong>Note: Please update globally installed package frequently. :innocent:	</strong>
- Installing from `pypi`
    - `pip install covid19-cli` (use pip for Python3)
    - Already installed ?
        - Update using `pip install -U covid19-cli`
        - see [How To Update Pip Package](https://stackoverflow.com/questions/4536103/how-can-i-upgrade-specific-packages-using-pip-and-a-requirements-file)
- Installing Manually :
    - `git clone https://github.com/asprazz/covid19-cli.git`
    - `cd covid19-cli`
        - Option 1:
            - (if windows) `pip install .`
            - (if linux/mac) `sudo pip install . -H`
        - Option 2:
            - (optional for linux users) if requires `chmod +x install.sh`
            - then run `scripts/./install.sh`


### Usage
- `covid19 -e` or `covid19 --emergency` : for printing emergency numbers
- `covid19 -h` or `covid19 --help` : for printing emergency numbers
- `covid19 {COUNTRY}` :
    - `covid19 World` : for whole World's status world is special keyword
    - `covid19 India` or `covid19 Ind` or `covid19 In` : for India's status
    - `covid19 USA` or `covid19 Us` : for Fetching Usa's status
    - `covid19 UK` or `covid19 GBR` : for Fetching UK's status
    - `covid19 World -a` or `covid19 World --all` : for Fetching all countries at once
    - `covid19 India -a` or `covid19 India --all` : for Fetching all states at once (alternative to `covid India -s mh`)
    - For fetching states in India (only for India)
        - `covid19 India -s maharashtra` or `covid19 Ind -s=maharashtra` : for both country's and state's status
        - `covid19 India -s mh` : will work the same way
        - `covid19 India -s all` or `covid19 In -s=ALL` : for all states of India
- See CODES.md (https://github.com/asprazz/covid19-cli/blob/master/CODES.md)
- See Documentation (https://github.com/asprazz/covid19-cli/blob/master/docs/)

### Contributing Guidelines
- Thank you for Showing interest in contributing to this project
- Please see https://github.com/asprazz/covid19-cli/blob/master/CONTRIBUTORS.md

#### Development
- Please, follow the contributing guidelines
- Fork the repository and clone it to your local environment
- Activate environment if any (
    [`venv`](https://docs.python.org/3/library/venv.html)
    or [`conda`](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html)
    for more details
  )
- Running `covid19`
  - Running locally
      - `cd covid19-cli`
      - `python covid19/__main__.py`
  - Installing globally from your local repository
      - `cd covid19-cli`
          - Option 1:
              - (if windows) `pip install .`
              - (if linux/mac) `sudo pip install . -H`
          - Option 2:
              - (optional for linux users) if requires `chmod +x install.sh`
              - then run `scripts/./install.sh`
          - option 3:
              - `python setup.py install`
- Fix :wrench: something broken or Build :hammer: something interesting
- Don't forget to target `develop` branch only

### Contributing Guidelines
- Thank you for Showing interest in contributing to this project
- Please see https://github.com/asprazz/covid19-cli/blob/master/CONTRIBUTORS.md

#### Development
- Please, follow the contributing guidelines
- Fork the repository and clone it to your local environment
- Activate environment if any (
    [`venv`](https://docs.python.org/3/library/venv.html)
    or [`conda`](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html)
    for more details
  )
- Running `covid19`
  - Running locally
      - `cd covid19-cli`
      - `python covid19/__main__.py`
  - Installing globally from your local repository
      - `cd covid19-cli`
          - Option 1:
              - (if windows) `pip install .`
              - (if linux/mac) `sudo pip install . -H`
          - Option 2:
              - (optional for linux users) if requires `chmod +x install.sh`
              - then run `scripts/./install.sh`
          - option 3:
              - `python setup.py install`

### Error reports
- First of all thank you.
- https://github.com/asprazz/covid19-cli/issues


### Special Thanks To Covid19India
- Built on API by :  https://api.covid19india.org/
- For more details : https://github.com/covid19india/api
- Also Visit: https://www.covid19india.org/


### Contributors
- Ankush Patil (https://github.com/asprazz)
- Suraj Kulkarni (https://github.com/KulkarniSuraj)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [1.0.9]

### Added
- [x] Added All Country View

### Changed
- [x] Dev: parser
- [x] Updated readme

## [1.0.8]

### Added
- [x] Fix Bug in printing all states info
- [x] Fixes #17 and #19

### Changed

## [1.0.7] -  2020-04-12

### Added
- [x] Added comma in numbers to improve readability Issue #14
### Changed
- [x] Changed import structure

## [1.0.5][1.0.6] - 2020-04-09

### Added
- Added capability for parsing all states at a single time
- Added world status fetcher
- Added loader while waiting for request
- Added capability to fetch single country
### Changed
- Changed menu
- Changed Readme.md added proper guidelines on usage


covid19/fetcher.py
import pprint
import time
from datetime import datetime

from covid19.helpers import send_request

# api.covid19india.org
DATA_API = "https://api.covid19india.org/data.json"
DISTRICT_WISE_API = "https://api.covid19india.org/state_district_wise.json"

# NovelCovid/API for all world
WORLD_API = "https://corona.lmao.ninja/v2/all"
COUNTRIES_API = "https://corona.lmao.ninja/v2/countries"


def fetch_india_status(state=False):
    try:
        DATA = send_request(DATA_API)

        if not state:
            # for India stateName is Total
            state = "Total"
        elif state.lower() == "all":
            # if data for every state requested
            return DATA['statewise'][1:]

        for details in DATA['statewise']:
            if details['state'].lower() == state.lower() or \
               details['statecode'].upper() == state.upper():
                return details

    except Exception as e:
        return None


def fetch_world_details():
    try:
        DATA = send_request(WORLD_API)
        # match data according global parser object
        WORLD_DETAILS = {}
        WORLD_DETAILS['state'] = "World"
        WORLD_DETAILS['lastupdatedtime'] = time.strftime(
            '%d/%H/%Y %H:%M',
            time.gmtime(DATA['updated']/1000)
        )
        WORLD_DETAILS['confirmed'] = DATA['cases']
        WORLD_DETAILS['active'] = DATA['active']
        WORLD_DETAILS['recovered'] = DATA['recovered']
        WORLD_DETAILS['deaths'] = DATA['deaths']

        WORLD_DETAILS['deltaconfirmed'] = DATA['todayCases']
        WORLD_DETAILS['detlaactive'] = 0
        WORLD_DETAILS['deltarecovered'] = 0
        WORLD_DETAILS['deltadeaths'] = DATA['todayDeaths']

        WORLD_DETAILS['affectedCountries'] = DATA['affectedCountries']

        return WORLD_DETAILS

    except Exception as e:
        return None


def fetch_country_details(country):
    try:
        COUNTRIES_DATA = send_request(COUNTRIES_API)
        DATA = {}

        if country != "all":
            for COUNTRY_DATA in COUNTRIES_DATA:
                if COUNTRY_DATA['country'].lower() == country.lower():
                    DATA = COUNTRY_DATA
                    break

                if 'countryInfo' in COUNTRY_DATA:
                    if 'iso2' in COUNTRY_DATA['countryInfo'] and \
                       COUNTRY_DATA['countryInfo']['iso2'] is not None:
                        if COUNTRY_DATA['countryInfo']['iso2'].lower() == country.lower():
                            DATA = COUNTRY_DATA
                            break

                    if 'iso3' in COUNTRY_DATA['countryInfo'] and \
                       COUNTRY_DATA['countryInfo']['iso3'] is not None:
                        if COUNTRY_DATA['countryInfo']['iso3'].lower() == country.lower():
                            DATA = COUNTRY_DATA
                            break
            COUNTRY_DETAILS = {}
            COUNTRY_DETAILS['state'] = DATA['country']
            COUNTRY_DETAILS['lastupdatedtime'] = time.strftime(
                '%d/%H/%Y %H:%M',
                time.gmtime(DATA['updated']/1000)
            )
            COUNTRY_DETAILS['confirmed'] = DATA['cases']
            COUNTRY_DETAILS['active'] = DATA['active']
            COUNTRY_DETAILS['recovered'] = DATA['recovered']
            COUNTRY_DETAILS['deaths'] = DATA['deaths']

            COUNTRY_DETAILS['deltaconfirmed'] = DATA['todayCases']
            COUNTRY_DETAILS['detlaactive'] = 0
            COUNTRY_DETAILS['deltarecovered'] = 0
            COUNTRY_DETAILS['deltadeaths'] = DATA['todayDeaths']

            COUNTRY_DETAILS['critical'] = DATA['critical']

            return COUNTRY_DETAILS
        else:
            COUNTRIES_DETAILS = []
            for DATA in COUNTRIES_DATA:
                COUNTRY_DETAILS = {}
                COUNTRY_DETAILS['state'] = DATA['country']
                COUNTRY_DETAILS['lastupdatedtime'] = time.strftime(
                    '%d/%H/%Y %H:%M',
                    time.gmtime(DATA['updated']/1000)
                )
                COUNTRY_DETAILS['confirmed'] = DATA['cases']
                COUNTRY_DETAILS['active'] = DATA['active']
                COUNTRY_DETAILS['recovered'] = DATA['recovered']
                COUNTRY_DETAILS['deaths'] = DATA['deaths']

                COUNTRY_DETAILS['deltaconfirmed'] = DATA['todayCases']
                COUNTRY_DETAILS['detlaactive'] = 0
                COUNTRY_DETAILS['deltarecovered'] = 0
                COUNTRY_DETAILS['deltadeaths'] = DATA['todayDeaths']

                COUNTRY_DETAILS['critical'] = DATA['critical']
                COUNTRIES_DETAILS.append(COUNTRY_DETAILS)
            return COUNTRIES_DETAILS

    except Exception as e:
        return None


def fetch_sos_details():
    print("_______Helpline Numbers___________\n")
    print("1. Andra Pradesh : 0866-2410978")
    print("2. Arunachal Pradesh : 9436055743")
    print("3. Assam : 6913347770")
    print("4. Bihar : 104")
    print("5. Chhattisgarh : 104")
    print("6. Goa : 104")
    print("7. Gujrat : 104")
    print("8. Haryana : 8558893911")
    print("9. Himachal Pradesh : 104")
    print("10. Jharkhand : 104")
    print("11. Karnataka : 104")
    print("12. Kerala : 0471-2552056")
    print("13. Madhya Pradesh : 104")
    print("14. Maharashtra : 020-26127394")
    print("15. Manipur : 3852411668")
    print("16. Meghalaya : 108")
    print("17. Mizoram : 102")
    print("18. Nagaland : 7005539653")
    print("19. Odisha : 9439994859")
    print("20. Punjab : 104")
    print("21. Rajasthan : 0141-2225624")
    print("22. Sikkim : 104")
    print("23. Tamil Nadu : 044-29510500")
    print("24. Telangana : 104")
    print("25. Tripura : 0381-2315879")
    print("26. Uttarakhand : 104")
    print("27. Uttar Pradesh  : 18001805145")
    print("28. West Bengal : 1800313444222, 03323412600")
    print("")
    print("1.Andaman and Nicobar Islands : 03192-232102")
    print("2. Chandigarh : 9779558282")
    print("3. Dadra and Nagar Haveli and Daman & Diu 1 : 104")
    print("4. Delhi : 011-22307145")
    print("5. Jammu & Kashmir  : 01912520982, 0194-2440283")
    print("6. Ladakh : 01982256462")
    print("7. Lakshadweep : 104")
    print("8. Puducherry : 104")
    print("")
    print("")
    print("______________HELPFUL LINKS______________")
    print("https://www.mohfw.gov.in/pdf/coronvavirushelplinenumber.pdf")
    print("https://www.mohfw.gov.in/")
    print("https://www.who.int/emergencies/diseases/novel-coronavirus-2019")
    print("https://www.cdc.gov/coronavirus/2019-ncov/faq.html")
    print("https://coronavirus.thebaselab.com/")
    print("")
    print("")
